src/data.ts

```diff
@@ -12,15 +13,31 @@ export interface DataFormat {
    * For Specific date formats can be provided (e.g., `{foo: 'date:"%m%d%Y"'}`), using the [d3-time-format syntax](https://github.com/d3/d3-time-format#locale_format). UTC date format parsing is supported similarly (e.g., `{foo: 'utc:"%m%d%Y"'}`). See more about [UTC time](timeunit.html#utc)
    */
   parse?: 'auto' | object;
+  /**
+   * Type of input data: `"json"`, `"csv"`, `"tsv"`.
+   * The default format type is determined by the extension of the file URL.
+   * If no extension is detected, `"json"` will be used by default.
+   */
+  type?: DataFormatType;
+}
 
+export interface CsvDataFormat extends DataFormatBase {
+  type?: 'csv' | 'tsv';
```

Comment:
Shouldn't this get the same JSDocs comment as the original type it overrides?

(So that the editor always get comment during autocompletion.)

Reply:
No. That's the nice thing about the change. Only supported properties are autocompleted. 

<img width="658" alt="screen shot 2017-09-10 at 13 39 56" src="https://user-images.githubusercontent.com/589034/30252794-910acc9a-962d-11e7-8377-477b5f399ea8.png">

Reply:
I mean adding `type?: 'csv' | 'tsv';` here is good .
But currently, if you mouseover "type" in the editor, you won't get JSDocs comment like you hover other properties such as "format". 

Thus, I wonder if copying JSDocs from the original to all specific `type` you override would include JSDocs comment for `type` when users hover?

Reply:
Done